### PR TITLE
WIP:iox-#1394 fix axivion reported violations for relative_pointer

### PR DESCRIPTION
**Pull request by saif-at-github**
Signed-off-by: shaik saifulla <shaik.saifulla@apex.ai>

## Pre-Review Checklist for the PR Author

1. [x] Code follows the coding style of [CONTRIBUTING.md][contributing]
1. [x] Tests follow the [best practice for testing][testing]
1. [x] Changelog updated [in the unreleased section][changelog] including API breaking changes
1. [x] Branch follows the naming format (`iox-123-this-is-a-branch`)
1. [x] Commits messages are according to this [guideline][commit-guidelines]
    - [x] Commit messages have the issue ID (`iox-#123 commit text`)
    - [x] Commit messages are signed (`git commit -s`)
    - [x] Commit author matches [Eclipse Contributor Agreement][eca] (and ECA is signed)
1. [x] Update the PR title
   - Follow the same conventions as for commit messages
   - Link to the relevant issue
1. [x] Relevant issues are linked
1. [x] Add sensible notes for the reviewer
1. [x] All checks have passed (except `task-list-completed`)
1. [x] All touched (C/C++) source code files from `iceoryx_hoofs` are added to `./clang-tidy-diff-scans.txt`
1. [x] Assign PR to reviewer

[commit-guidelines]: https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html
[eca]: http://www.eclipse.org/legal/ECA.php
[contributing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/CONTRIBUTING.md#coding-style
[testing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/concepts/best-practice-for-testing.md
[changelog]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/release-notes/iceoryx-unreleased.md

## Checklist for the PR Reviewer

- [ ] Commits are properly organized and messages are according to the guideline
- [ ] Code according to our coding style and naming conventions
- [ ] Unit tests have been written for new behavior
- [ ] Public API changes are documented via doxygen
- [ ] Copyright owner are updated in the changed files
- [ ] All touched (C/C++) source code files from `iceoryx_hoofs` have been added to `./clang-tidy-diff-scans.txt`
- [ ] PR title describes the changes

## Post-review Checklist for the PR Author

1. [ ] All open points are addressed and tracked via issues

## References

- Closes **TBD**

### CODE REVIEW

File: iceoryx_hoofs/include/iceoryx_hoofs/internal/relocatable_pointer/relative_pointer.inl [hunk 1]
```diff
@@ -24,23 +24,23 @@ namespace iox
...
-    : BaseRelativePointer(offset, id)
-{
-}
-
-template <typename T>
-inline RelativePointer<T>::RelativePointer(ptr_t ptr) noexcept
-    : BaseRelativePointer(ptr)
-{
-}
+// template <typename T>
```
Comment (saif-at-github) [inline iceoryx_hoofs/include/iceoryx_hoofs/internal/relocatable_pointer/relative_pointer.inl:27]: commented to address the violation, 
'Rule A12-1-6 : Derived classes that do not need further explicit initialization and require
all the constructors from the base class shall use inheriting constructors.'

File: iceoryx_hoofs/include/iceoryx_hoofs/internal/relocatable_pointer/relative_pointer.hpp [hunk 2]
```diff
@@ -36,22 +36,24 @@ class RelativePointer : public BaseRelativePointer
...
+    using BaseRelativePointer::BaseRelativePointer;
+
     /// @brief default constructs a RelativePointer as a logical nullptr
     RelativePointer() noexcept = default;
 
     /// @brief constructs a RelativePointer pointing to the same pointee as ptr in a segment identified by id
     /// @param[in] ptr the pointer whose pointee shall be the same for this
     /// @param[in] id is the unique id of the segment
-    RelativePointer(ptr_t ptr, id_t id) noexcept;
+    //    RelativePointer(ptr_t ptr, id_t id) noexcept;
```
Comment (saif-at-github) [inline iceoryx_hoofs/include/iceoryx_hoofs/internal/relocatable_pointer/relative_pointer.hpp:47]: Rule A12-1-6 (required, implementation, automated)
Derived classes that do not need further explicit initialization and require
all the constructors from the base class shall use inheriting constructors.

File: iceoryx_hoofs/include/iceoryx_hoofs/internal/relocatable_pointer/relative_pointer.hpp [hunk 3]
```diff
@@ -36,22 +36,24 @@ class RelativePointer : public BaseRelativePointer
     using const_ptr_t = const T*;
     using ptr_t = T*;
 
+    using BaseRelativePointer::BaseRelativePointer;
```
Comment (saif-at-github) [inline iceoryx_hoofs/include/iceoryx_hoofs/internal/relocatable_pointer/relative_pointer.hpp:39]: This added code will be acceptable? let me know your comments on this? 'Rule A12-1-6'

Comment (elBoberido) [reply iceoryx_hoofs/include/iceoryx_hoofs/internal/relocatable_pointer/relative_pointer.hpp:39]: @saif-at-github do you know if doxygen shows the ctors inherited this way? If yes, then this is totally fine. If not, we need to think how we can make this visible to the user.

cc @MatthiasKillat

**Pull request closed without merging**